[enh] add test.yamllint - lint yaml files

return42 opened the pull request:
Usage::

    make test.yamllint

    ./manage test.yamllint

test.yamllint is also added to the `test` makefile target.

return42 committed:
[enh] add test.yamllint - lint yaml files

Usage::

    make test.yamllint

    ./manage test.yamllint

test.yamllint is also added to the `test` makefile target.

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/syntaxerror_settings.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/user_settings.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/user_settings_keep_only.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/user_settings_simple.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/user_settings_remove.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] tests/unit/settings/user_settings_remove2.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] searx/settings_robot.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

return42 committed:
[yamllint] searx/settings.yml

Fixed messages reported by::

    make test.yamllint

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>

dalf commented on manage:
```diff
@@ -29,6 +29,11 @@ pylint.FILES() {
 
+YAMLLINT_FILES=()
+while IFS= read -r line; do
+   YAMLLINT_FILES+=("$line")
```
It should ignore files defined by `.gitignore`.
More into details: `make themes` will install different NPM packages, some of them includes .yml files which don't follow the yamllint format.

Perhaps `git ls-files '*.yml'` ?

return42 replied:
fixed to: `git ls-files './tests/*.yml' './searx/*.yml'` / see [9ce7dc6f](https://github.com/searxng/searxng/pull/105/commits/9ce7dc6f2d4b5dd14b0f0ff04ff9f7a333595ef9#diff-181229424893bb65d94a74c2132b8b9e5adfe851464fdb5cb9f49e8a8204be7bR35)

return42 committed:
[emacs] flycheck - activate yamllint and use .yamllint.yml

Open YAML file and use:

  M-x flycheck-mode

Since we use yamllint from local/py3/bin, creating a virtualenv is a
prerequisite:

  make pyenv

Signed-off-by: Markus Heiser <markus.heiser@darmarit.de>
